### DISCUSSION TITLE
[Enhancement] CollectionView Body

### DISCUSSION THREAD

**Discussion by angelru** [xamarin/Xamarin.Forms#11338] [General]
## Summary
We have header and footer in CollectionView, and we can put any view there, but in ItemTemplate we only have the view that repeats in the content, it would be nice to have a main content that was only for the item source (not repeating for Every item ) Let's suppose that CollectionView has a header that loads a view that requires a call to my api, there I can show an activity indicator only in the header and if then I want to later load my content in ItemTemplate? I don't have a Body that can load a container view for repeating items, this is done by instagram, first load the name and photo with your followers, then load the user photos, and show 2 activity indicator (one for the profile and another for the photos).

I hope I have explained well.

```
<CollectionView>
      <CollectionView.Body>
     <StackLayout>
                <CollectionView.ItemTemplate>
                    <DataTemplate>
                        <Grid>
                         </Grid>
                    </DataTemplate>
                </CollectionView.ItemTemplate>
    </StackLayout>
     <CollectionView.Body>
</CollectionView>
```

**Comment by EmilAlipiev** [comment]
why dont you just put the stacklayout around your collectionview like below

```
 <StackLayout Orientation=Vertical>
     <HeaderView "/>
    <ActivityIndicator  IsVisible=Bind to collectionview visibility or a bool parameter in viewmodel handles reverse visibility to CW visibility />
   <CollectionView IsVisible=Inverse Visibility as  ActivityIndicator  />
  </StackLayout>
```

**Comment by angelru** [comment]
@EmilAlipiev 
Suppose I'm going to use something like this: [TaskLoaderView](https://github.com/roubachof/Sharpnado.TaskLoaderView)

And I want the scroll to be integer to the whole page, not want the header remains fixed.
So in CollectionView in the header, I load content with my api (other than itemsource).

Then a few seconds later, I load the ItemSource content, both in the header and ItemSorce I want activityindicator, because it is independent content.

For example, the first content is simple data (only a few seconds) and the content of the item is many photos (it takes more time), it is a good experience for the user to be able to show the simple data first and then the photos. without hiding the CollectionView

```
<CollectionView>
<CollectionView.Header>
<customViews:TaskLoaderView>
        <Grid>
        </Grid>
</customViews:TaskLoaderView>
    </CollectionView.Header>
      <CollectionView.Body>
   <customViews:TaskLoaderView>
                <CollectionView.ItemTemplate>
                    <DataTemplate>
                        <Grid>
                         </Grid>
                    </DataTemplate>
                </CollectionView.ItemTemplate>
  </customViews:TaskLoaderView>
     <CollectionView.Body>
</CollectionView>
```

**Comment by PureWeen** [comment]
What about 
https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/collectionview/emptyview

?

**Comment by angelru** [comment]
@PureWeen  it doesn't quite fit my needs ... but "I can make a fix"

**Comment by angelru** [comment]
@PureWeen  It is difficult for me to create a design with parallax effect with collectionview, with body, it would be possible
